arangod/Cluster/ClusterInfo.h

```diff
@@ -72,24 +71,27 @@ class CollectionWatcher
 
     _agencyCallback = std::make_shared<AgencyCallback>(
         ac, where,
-        [this](VPackSlice const& result) {
-          if (result.isNone()) {
-            _present.store(false);
+        [self = weak_from_this()](VPackSlice const& result) {
```

Comment:
I think `weak_from_this` is C++17, is this allowed in 3.6? If not, we probably have to make a copy of the shared_ptr for 3.6.

Reply:
3.6 is C++17.